lib/entities/quote/DutchQuote.ts

```diff
@@ -358,15 +358,19 @@ export abstract class DutchQuote<T extends DutchQuoteRequest> implements IQuote
   }
 
   // Returns the number of gas units extra required to execute this quote through UniswapX
-  static getGasAdjustment(classicQuote: ClassicQuote): BigNumber {
+  static getGasAdjustment(classicQuote: ClassicQuote, adjustmentBps?: number): BigNumber {
     let result = BigNumber.from(0);
 
     // fill contract must unwrap WETH output tokens
     if (classicQuote.request.info.tokenOut === NATIVE_ADDRESS) {
       result = result.add(WETH_UNWRAP_GAS);
     }
 
-    return result.add(UNISWAPX_BASE_GAS);
+    result = result.add(UNISWAPX_BASE_GAS);
+    if (adjustmentBps) {
+      result = result.mul(BPS - adjustmentBps).div(BPS);
```

Comment:
Just to double check that we want to reduce the gas (not add a buffer), right? By default I think of a buffer as additive. Could you mention that it reduces the gas by this buffer in the comments?